go/tools/builders/pack.go

```diff
@@ -330,3 +337,36 @@ func appendFiles(goenv *env, archive string, files []string) error {
 	args = append(args, files...)
 	return goenv.runCommand(args)
 }
+
+type fileMatcher func(name string) bool
+var fileNotFound = errors.New("file not found in archive")
+
+func readFileInArchive(match fileMatcher, archive string) ([]byte, error) {
```

Comment:
Let's change the signature to `func readFileInArchive(archive, name string) (io.ReadCloser, error)`.

`fileMatcher` seems like overkill to me. For nogo facts, we can just use a constant file name.

`io.ReadCloser` should probably be used instead of `[]byte`, since there's no upper bound on file size. The caller can use `ioutil.ReadAll` if they want. You can use `io.LimitReader` to restrict reading to a single file within the archive.